apps/cnquery/cmd/plugin.go

```diff
@@ -123,6 +123,17 @@ func (c *cnqueryPlugin) RunQuery(conf *run.RunQueryConfig, runtime *providers.Ru
 		out.WriteString("[")
 	}
 
+	// anyResultFailed is a flag that will be switched on if any query result failed,
+	// if the flag `exit-1-on-failure` is provided and anyResultFailed is true, we
+	// will exit the program with the exit code `1`
+	var anyResultFailed = false
+	// we defer this check since we want it to be the last thing to be evaluated
+	defer func() {
+		if conf.GetExit_1OnFailure() && anyResultFailed {
+			os.Exit(1)
```

Comment:
if we exit here, we might not run anything that follows `RunQuery`, I think we should just return an err here and let the caller above handle this